src/runtimes/rego/stack_pre.rs

```diff
@@ -0,0 +1,56 @@
+use anyhow::Result;
+
+use crate::policy_evaluator::RegoPolicyExecutionMode;
+use crate::policy_evaluator_builder::EpochDeadlines;
+
+/// This struct allows to follow the `StackPre -> Stack`
+/// "pattern" also for Rego policies.
+///
+/// However, Rego policies cannot make use of `wasmtime::InstancePre`
+/// to reduce the instantiation times. That happens because all
+/// Rego WebAssembly policies import their Wasm Memory from the host.
+/// The Wasm Memory is defined inside of a `wasmtime::Store`, which is
+/// something that `wasmtime::InstnacePre` objects do not have (rightfully!).
+///
+/// However, Rego Wasm modules are so small thta instantiating them from scratch
+/// is already a cheap operation.
+#[derive(Clone)]
+pub(crate) struct StackPre {
+    engine: wasmtime::Engine,
+    module: wasmtime::Module,
+    epoch_deadlines: Option<EpochDeadlines>,
+    pub entrypoint_id: i32,
+    pub policy_execution_mode: RegoPolicyExecutionMode,
+}
+
+impl StackPre {
+    pub(crate) fn new(
+        engine: wasmtime::Engine,
+        module: wasmtime::Module,
+        epoch_deadlines: Option<EpochDeadlines>,
+        entrypoint_id: i32,
```

Comment:
Nit: If we are using always the same entrypoint `0` for now, maybe we can set it inside new() and not accept it as an argument.

Reply:
I prefer to leave the arguments around, because if you put a non-zero value in there everything will work as expected. The only place where the entrypoint is hard coded is the builder.